### PR TITLE
Simplify showcased client code

### PR DESCRIPTION
**Pull request by manishrjain**
Don't need the done channel because messages channel is being closed by the call to client.Fetch.

### CODE REVIEW

**Comment by emersion** [comment]
Can you do the same thing for `mailboxes`?

**Comment by manishrjain** [comment]
Done. Please have another look!

File: README.md [hunk 1]
```diff
@@ -115,20 +112,18 @@ func main() {
 	seqset, _ := imap.NewSeqSet("")
 	seqset.AddRange(mbox.Messages - 3, mbox.Messages)
 
-	messages := make(chan *imap.Message)
-	done = make(chan error, 1)
+	messages := make(chan *imap.Message, 10)
```
Comment (emersion) [inline README.md]: I didn't realized when I first looked that you used a buffered channel here. Is there a particular reason?

The `for msg := range messages` loop should process messages as soon as they arrive.

**Comment by manishrjain** [comment]
In an unbuffered channel, each put into channel would block until there's an exact get from the channel. It basically makes both put and get synchronous. So, instead of producer working independently, concurrently from the receiver, they block on each other -- which isn't a great design.

Honestly, I think there are very few uses of unbuffered channels -- and they're very prone to deadlocks.

File: README.md (hunk 1 above)
Comment (manishrjain) [inline README.md]: Answered directly in the PR.

**Comment by emersion** [comment]
Okay, this makes sense, thanks for explaining!

LGTM

**Pull request merged**